app/assets/javascripts/member-facing/redirectors.js

```diff
@@ -0,0 +1,43 @@
+const RegisterMemberRedirector = {
+  attemptRedirect:  function(followUpUrl, member) {
+
+    if(typeof(member) !== 'object') {
+      member = window.champaign.personalization.member;
+    }
+
+    if ( shouldMemberRegister() ) {
+      redirectToRegistration();
+      return true;
+    } else {
+      return false;
+    }
+
+    function shouldMemberRegister() {
+      return !member.registered;
+    }
+
+    function redirectToRegistration() {
```

Comment:
@osahyoun I like that syntax too, but the arrow syntax only works if you assign a function to a variable, here I'm defining the functions after using them, so I believe it doesn't apply to this case. Let me know if I'm wrong.